extide/gradle/src/org/netbeans/modules/gradle/api/execute/GradleDistributionManager.java

```diff
@@ -308,6 +313,46 @@ public GradleDistribution defaultDistribution() {
         return ret;
     }
 
+    /**
+     * Lists all the {@link GradleDistribution}s available on the Gradle Home
+     * of this distribution manager. It looks for the <code>$GRADLE_HOME/wrapper/dists</code>
+     * directory for already downloaded distributions.
+     * @return the list of available Gradle distributions from the Gradle Home.
+     * @since 2.10
+     */
+    public List<GradleDistribution> availableLocalDistributions() {
+        List<GradleDistribution> ret = new ArrayList<>();
+        File dists = new File(gradleUserHome, "wrapper/dists"); //NOI18N
+        if (dists.isDirectory()) {
+            try {
+                Files.walkFileTree(dists.toPath(), new SimpleFileVisitor<Path>() {
+                    @Override
+                    public FileVisitResult visitFile(Path f, BasicFileAttributes attrs) throws IOException {
+                        String fname = f.getFileName().toString();
+                        Matcher m = DIST_VERSION_PATTERN.matcher(fname);
+                        if (m.matches()) {
```

Comment:
this looks for `xxxx-gradle-version.zip` in `~/.gradle/wrapper/dists`, and uses its unpacked form, right ?

Reply:
well, the structure below ```~/.gradle/wrapper/dists``` is ```gardle-<version>-<bin|all>/<hash>/gradle-<version>/``` and in the ```<hash>``` folder there is the distribution zip as well.

Reply:
oh ! Sorry. Then I'd suggest to stop at level 2 in the file walk (f.getNameCount() > dists.getNameCount() + 2)